### ISSUE TITLE
ms sql replica won't build

### ISSUE THREAD

**Issue by brantleyp1** [terraform-aws-modules/terraform-aws-rds#268]
I've hit a wall trying to launch an RDS instance and a replica instance using SQL server 14.00xxx. I can launch the primary node without issue, but the replica fails with:

`Error: Error creating DB Instance: InvalidParameterCombination: An option group can't be specified when creating or modifying a Read Replica for the sqlserver-ee engine.`

I've set create_db_option_group to false, I've tried true. I've tried create_db_parameter_group to false and true. Currently I've hacked up the module to comment out any mention of an option group and it still fails. I'm not sure if I've stumbled upon a bug or have a serious layer 8 issue, but I cannot navigate past it. Any help available would be greatly appreciated.

My code for the RDSs (note, I've left commented out lines as is and added a few notes for clarity for this ticket):

```
module "db" {
  source = "terraform-aws-modules/rds/aws"

  identifier        = "${var.namespace}-${var.workload_name}-rds"
  engine            = var.rds_db_engine  ## sql-server-ee-14.0
  engine_version    = var.rds_db_engine_version  ## 14.00.3294.2.v1
  instance_class    = var.rds_config[var.operational_band].instance_type ## db.m5.xlarge
  allocated_storage = var.rds_config[var.operational_band].volume_size ## 100
  storage_type      = var.rds_storage_type ## io1
  iops              = var.rds_config[var.operational_band].iops ## 1000
  storage_encrypted = true

  //name     = "${var.namespace}-${var.workload_name}-rds"
  username = var.rds_username
  password = random_string.rds_password.result
  port     = var.rds_port

  maintenance_window = "Sat:23:00-Sun:03:00"
  backup_window      = "03:00-06:00"

  # set to 1 to allow replicas, 0 if not for faster db build
  backup_retention_period = 1

  multi_az = true

  subnet_ids             = data.aws_subnet_ids.private.ids
  vpc_security_group_ids = data.aws_security_groups.storage.ids
  tags = merge(
    local.rds_tags,
    { ops-instancename = "${var.namespace}.${var.workload_name}-rds"
      Name                  = "${var.namespace}.${var.workload_name}-rds"
  })

  timeouts = {    ## I have and have not tried without this option
    create = "80m"
    delete = "80m"
    update = "80m"
  }

  parameters = tolist(local.default_parameters[var.rds_db_family]) ## sets force_ssl = true ## I have and have not tried without this option

  # DB option group
  major_engine_version = "14.00"

  license_model = "license-included"

  final_snapshot_identifier = "${var.namespace}-${var.workload_name}-final-snapshot"
  create_db_parameter_group = false
  //create_db_option_group    = false  ## have tried every combination
  //option_group_name = "${var.workload_name}-${var.rds_db_engine}-option-group". ## same, have set and unset/removed this
  deletion_protection       = true
  skip_final_snapshot = true

  depends_on = [
    random_string.rds_password,
  ]
}

module "replica_db" {
  source = "terraform-aws-modules/rds/aws"

  identifier        = "${var.namespace}replica"
  engine            = var.rds_db_engine
  engine_version    = var.rds_db_engine_version
  instance_class    = var.rds_config[var.operational_band].instance_type
  allocated_storage = var.rds_config[var.operational_band].volume_size
  storage_type      = var.rds_storage_type
  iops              = var.rds_config[var.operational_band].iops
  storage_encrypted = true
  //option_group_name = module.db.this_db_option_group_id
  //option_group_name = "${var.workload_name}-${var.rds_db_engine}-option-group"

  multi_az            = false                         
  replicate_source_db = module.db.this_db_instance_id 
  username            = ""                            
  password            = ""                            
  port                = var.rds_port

  maintenance_window = "Sat:23:00-Sun:03:00"
  backup_window      = "03:00-06:00"

  backup_retention_period = 0

  major_engine_version = "14.00"

  license_model = "license-included"

  vpc_security_group_ids = data.aws_security_groups.storage.ids

  tags = merge(
    local.rds_tags,
    { ops-instancename = "${var.namespace}.${var.workload_name}-replica-rds"
      Name                  = "${var.namespace}.${var.workload_name}-replica-rds"
  })

  deletion_protection       = false
  create_db_subnet_group    = false
  //create_db_option_group    = false
  create_db_parameter_group = false
  skip_final_snapshot = true

  depends_on = [
    module.db,
  ]
}
```

If I comment out the major_engine_version option, I get another error that leads me to believe it still is trying to build the option group:
`Error: Error creating DB Option Group: InvalidParameterValue: The parameter MajorEngineVersion must be provided and must not be blank.`

**Comment by brantleyp1** [comment]
Reading through the issues, I think #188 and #230 are related. I tried to replicate the code commit changes from #188 but got the same results. I also attempted to put a dummy option group name as well as setting the create_db_option_group to false but instead it threw an error that the dummy group did not exist.

**Comment by brantleyp1** [comment]
After a ton of trial and error, I was able to get a working code by setting:

1. In module.db - `create_db_option_group    = false`
2. In module.db - iops and storage_type are commented out
3. In module.replica_db - `option_group_name = module.db.this_db_option_group_id` had to be set
4. In module.replica_db - iops and storage_type were commented out, allocated_storage had to be set
5. In module.replica_db - I set the timeouts to 80m just to be safe, the first attempt the replica timed out at 40m.

Some oddities:

I'm running this in a testing environment so I'm intentionally destroying everything if everything isn't successfully built. But this results in a loop where because I had not created the db_option_group it would not allow me to play/apply/destroy because count was set to 0 in the replica_db module for option group. i.e. I ran apply, it created the other infrastructure and the primary db instance, but timed out during the replica. If I tried to then re-apply, destroy, or even just plan, it would error out. If I target the module.db then plan/apply/destroy will succeed; to destroy everything I have to target module.db first, which destroys replica, then comment out those sections, then destroy the rest. 

I'm not sure if this is a corner case, or if I'm using something in a way unexpected, I'm very green on terraform.
